python/python/lancedb/table.py

```diff
@@ -95,6 +95,8 @@ def _sanitize_data(
             data = _to_record_batch_generator(
                 data.data.to_batches(), schema, metadata, on_bad_vectors, fill_value
             )
+    if isinstance(data, LanceModel):
+        raise TypeError("Pass the data as a list : tbl.add([LanceModel_1(),..])")
```

Comment:
I'd also recommend adding a test that we get the expected error here. That way it doesn't get broken later when someone refactors this.